packages/server-wallet/src/protocols/close-channel.ts

```diff
@@ -216,6 +217,22 @@ const defundIntoLedger = (ps: ProtocolState): RequestLedgerDefunding | false =>
     assetHolderAddress: checkThat(ps.app.latest.outcome, isSimpleAllocation).assetHolderAddress,
   });
 
+/**
+ * If the channel is directly funded, then ensure none of its allocation items are other
+ * channels being funded by this channel (e.g., if it is a ledger channel). This should
+ * cause the protocol to "pause" / "freeze" until no channel depends on this channel
+ * for funding.
+ */
+const ensureAllAllocationItemsAreExternalDestinations = (ps: ProtocolState): boolean =>
+  (!isDirectlyFunded(ps.app) && !isFakeFunded(ps.app)) ||
```

Comment:
```suggestion

```
I believe this line is equivalent to `if (true) then` since direct and fake funding are mutually exclusive ?

Reply:
It's only true if it's Ledger funded I think, otherwise false

Reply:
Yeah ok, but I think it is still unnecessary given the next line?

Also, why not use your `isLedgerFunded` helper here?

Reply:
The original intent was for this to only apply to channels which were directly funded. However we still have this legacy "fake" funded channel type, so I added that, and we only have three types in the codebase now, so I suppose it's equivalent to checking if it's not ledger funded but then it kind of defeats the point.

I think the suggestion of just ensuring no internal destination exists inside the outcome for any channel before closing is reasonable.